### ISSUE TITLE
adding support for F(Q) in sasmodels (Trac #491)

### ISSUE THREAD

**Issue by butlerpd** [SasView/sasmodels#104]
I(Q) is proportional to F(Q)^2^ Hence our models return that.  However there are times when it may be preferable to have access to the F(Q).  For example it is required for the beta approximation (increasing the range of applicability for using most current interaction potentials) or if one wanted to use these as a library for building P(Q) for more complex shapes.

Providing a method to access the underlying F(Q) thus would be very useful.

Migrated from http://trac.sasview.org/ticket/491
```json
{
    "status": "assigned",
    "changetime": "2019-03-02T00:25:12",
    "_ts": "2019-03-02 00:25:12.521354+00:00",
    "description": "I(Q) is proportional to F(Q)^2^ Hence our models return that.  However there are times when it may be preferable to have access to the F(Q).  For example it is required for the beta approximation (increasing the range of applicability for using most current interaction potentials) or if one wanted to use these as a library for building P(Q) for more complex shapes.\n\nProviding a method to access the underlying F(Q) thus would be very useful.",
    "reporter": "butler",
    "cc": "",
    "resolution": "",
    "workpackage": "Beta Approximation Project",
    "time": "2015-12-19T19:07:45",
    "component": "sasmodels",
    "summary": "adding support for F(Q) in sasmodels",
    "priority": "major",
    "keywords": "",
    "milestone": "sasmodels 1.0",
    "owner": "wojciech",
    "type": "enhancement"
}
```

**Comment by butlerpd** [comment]
Trac update at `2016/03/20 22:41:08`: **butler** changed milestone from "SasView 4.0.0" to "SasView Next Release +1"

**Comment by ajj** [comment]
Trac update at `2016/10/07 19:58:00`: **ajj** changed description from:

> I(Q) is proportional to F(Q)^2 Hence our models return that.  However there are times when it may be preferable to have access to the F(Q).  For example it is required for the beta approximation (increasing the range of applicability for using most current interaction potentials) or if one wanted to use these as a library for building P(Q) for more complex shapes.
> 
> Providing a method to access the underlying F(Q) thus would be very useful.

to:

> I(Q) is proportional to F(Q)^2^ Hence our models return that.  However there are times when it may be preferable to have access to the F(Q).  For example it is required for the beta approximation (increasing the range of applicability for using most current interaction potentials) or if one wanted to use these as a library for building P(Q) for more complex shapes.
> 
> Providing a method to access the underlying F(Q) thus would be very useful.

**Comment by ajj** [comment]
Trac update at `2016/10/07 19:58:27`: 
* **ajj** changed owner from "" to "wojciech"
* **ajj** changed status from "new" to "assigned"

**Comment by pkienzle** [comment]
Trac update at `2018/10/30 22:20:27`: 
* **pkienzle** commented:

Supported in beta approximation branch of sasmodels.

External scripts that use sasmodels directly can call `sasmodels.direct_model.call_Iq` for I(Q) or  `sasmodels.direct_model.call_Fq` for F(Q).  For a structure factor calculation via the product model `P@S`, users can call `calculator.results()` to access the intermediates.



* **pkienzle** changed component from "SasView" to "sasmodels"
* **pkienzle** changed milestone from "SasView Next Release +1" to "sasmodels 1.0"

**Comment by butlerpd** [comment]
Trac update at `2019/03/02 00:25:12`: **butler** changed workpackage from "SasModels Redesign" to "Beta Approximation Project"

**Comment by pkienzle** [comment]
Fixed in #90.
